src/provider/provider_os_memory.c

```diff
@@ -1146,7 +1146,10 @@ typedef struct os_ipc_data_t {
     int fd;
     size_t fd_offset;
     size_t size;
-    char shm_name[NAME_MAX]; // optional - can be used or not (see below)
+    // shm_name is a Flexible Array Member because it is optional and its size
+    // varies on the Shared Memory object name
+    size_t shm_name_len;
```

Comment:
soo... should we get rid of the `NAME_MAX` define now...?

Reply:
NAME_MAX is used in other places and it is defined in limits.h

Reply:
I guess my question is - should we still use it in other places...?

Reply:
I think yes. User should know the IPC handle size on the producer and on the consumer sides.

Reply:
@vinser52 this means that we should have an api "umfGetIPCHandleSize(ipc_handle)" ?